modules/environment/default.nix

```diff
@@ -147,8 +141,23 @@ in
       '';
       type = types.lines;
     };
+
+    environment.extraSetup = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Shell fragments to be run after the system environment has been created.
+        This should only be used for things that need to modify the internals
+        of the environment, e.g. generating MIME caches.
+        The environment being built can be accessed at $out.
+      '';
+    };
   };
 
+  imports = [
+    (mkRenamedOptionModule ["environment" "postBuild"] ["environment" "extraSetup"])
+  ];
```

Comment:
Can you move this to be before all the options?

Reply:
Done